### PR TITLE
Support impl generation in current source dir

### PR DESCRIPTION
**Pull request by nanjj**
For example: impl 'adm *targetAdmin' TargetAdminServer

### CODE REVIEW

**Comment by josharian** [comment]
Thanks!

go module support is likely to entirely change how impl resolves identifiers, so I'm a bit hesitant to go too much further on the existing code, at least until I have some idea what the future looks like. Let's put this on hold a little while, if that's ok.

**Comment by josharian** [comment]
Tracking issue for module support: https://github.com/golang/go/issues/24661

**Comment by morganhein** [comment]
I'm personally going to implement this change locally, as impl's usefulness for myself is drastically reduced w/o it. As it stands, impl actually introduces bugs into code due to forcing the package name: https://play.golang.com/p/TIdxp6yJKJJ

**Comment by sebastianst** [comment]
any news here?

**Comment by josharian** [comment]
I took another look at this. I now believe it is mostly orthogonal to modules (which still need support added). I'll add a few line-level review comments. And it definitely needs a test.

File: impl.go [hunk 1]
```diff
@@ -84,17 +84,23 @@ func findInterface(iface string, srcDir string) (path string, id string, err err
...
-	}
-	raw := f.Imports[0].Path.Value   // "io"
-	path, err = strconv.Unquote(raw) // io
-	if err != nil {
-		panic(err)
+		if !strings.Contains(iface, ".") {
+			path = "."
+			id = iface
+		} else {
+			return "", "", fmt.Errorf("unrecognized interface: %s", iface)
```
Comment (josharian) [inline impl.go]: If you reorder this and unindent some it'll be easier to read. Comment, too.

```go
// If we were given an identifier with no "." in it, and goimports couldn't resolve it,
// assume it is from the current package.
if strings.Contains(iface, ".") {
  return "", "", fmt.Errorf("........")
}
return ".", iface, nil
```

Then the else below can be removed and the contents unindented as well.

File: impl.go [hunk 2]
```diff
@@ -157,7 +163,9 @@ func (p Pkg) fullType(e ast.Expr) string {
 			// the type isn't exported, there's no point trying
 			// to implement it anyway.
 			if n.IsExported() {
```
Comment (josharian) [inline impl.go]: Combine the conditions.

```go
if n.IsExported() && p.Package.Dir != *flagSrcDir {
  ....
}
```

Also, update the comment above to explain why the second half of the condition is present.

**Comment by josharian** [comment]
I believe that the recently pushed fixes for #46 should address this. Please feel free to open (or ping) an issue if not. Thanks!

**Pull request closed without merging**